### DISCUSSION TITLE
How is longhorn locally developed?

### DISCUSSION THREAD

**Discussion by cyclingwithelephants** [longhorn/longhorn#3096] [Q&A] [answered]
Hi folks, 

In creating a local dev environment using k3d I stumbled upon a blocker in https://github.com/rancher/k3d/discussions/478#discussioncomment-1190481

This got me wondering, what tools are used for local longhorn development? So far I've tried k3d (with the above issue) and kind, which doesn't have sudo in its node image

**Comment by jenting** [comment, marked as answer]
Generally, we'll bootstrap VMs with 4 nodes (1 control plane node and 3 worker nodes).
Currently, Longhorn can't install on k3d/kind/... which are container as the base host OS.

**Comment by cyclingwithelephants** [comment]
Thanks for the answer!
